bee-api/bee-rest-api/src/endpoints/config.rs

```diff
@@ -152,24 +134,37 @@ impl RestApiConfigBuilder {
         let port = multi_addr
             .iter()
             .find_map(|x| if let Protocol::Tcp(port) = x { Some(port) } else { None })
-            .unwrap_or_else(|| panic!("Unsupported protocol"));
-        let public_routes: Box<[String]> = self
-            .public_routes
-            .unwrap_or_else(|| DEFAULT_PUBLIC_ROUTES.iter().map(|s| s.to_string()).collect())
-            .into_boxed_slice();
-        let allowed_ips: Box<[IpAddr]> = self
-            .allowed_ips
-            .unwrap_or_else(|| DEFAULT_ALLOWED_IPS.to_vec())
-            .into_boxed_slice();
+            .expect("unsupported protocol");
+
+        let jwt_salt = self.jwt_salt.unwrap_or_else(|| DEFAULT_JWT_SALT.to_string());
+
+        let public_routes = {
+            let routes = self
+                .public_routes
+                .unwrap_or_else(|| DEFAULT_PUBLIC_ROUTES.iter().map(ToString::to_string).collect());
+            RegexSet::new(routes.iter().map(|r| route_to_regex(r)).collect::<Vec<_>>())
+                .expect("invalid public route provided")
+        };
+
+        let protected_routes = {
+            let routes = self
+                .protected_routes
+                .unwrap_or_else(|| DEFAULT_PROTECTED_ROUTES.iter().map(ToString::to_string).collect());
+            RegexSet::new(routes.iter().map(|r| route_to_regex(r)).collect::<Vec<_>>())
+                .expect("invalid protected route provided")
+        };
+
         let feature_proof_of_work = self.feature_proof_of_work.unwrap_or(DEFAULT_FEATURE_PROOF_OF_WORK);
+
         let white_flag_solidification_timeout = self
             .white_flag_solidification_timeout
-            .unwrap_or(DEFAULT_WHITE_FLAG_SOLIDIFICATION_TIMEOUT);
+            .map_or_else(|| DEFAULT_WHITE_FLAG_SOLIDIFICATION_TIMEOUT, Duration::from_secs);
```

Comment:
```suggestion
            .map_or(DEFAULT_WHITE_FLAG_SOLIDIFICATION_TIMEOUT, Duration::from_secs);
```
because it's just a constant we don't need to be lazy here.

Reply:
true